### ISSUE TITLE
Softmax implementation in chapter 10 is wrong?

### ISSUE THREAD

**Issue by AlexAndorra** [pymc-devs/resources#59]
I wonder if the softmax regressions implementation in chapter 10 is right. In `Code 10.57` it reads:
```python
with pm.Model() as m_10_16:
    b = pm.Normal('b', 0., 5.)
    s2 = b*2
    s3 = b*3
    p_ = pm.math.stack([b, s2, s3])
    obs = pm.Categorical('career', p=pm.math.sigmoid(p_), observed=career)
    trace_10_16 = pm.sample(1000, tune=1000)
```
but I think this should be:
```python
    p_ = pm.math.stack([0, s2, s3])
    obs = pm.Categorical('career', p=tt.nnet.softmax(p_), observed=career)
```
Otherwise, the model is over-parametrized, isn't it? You just need 2 categories to infer the respective probabilities of all 3.
Also, I think the link function should be softmax, not sigmoid (this is also the case in `Code 10.58`).

I can send a PR if these are indeed mistakes. Yours truly 😉

**Comment by junpenglao** [comment]
I think we are following the original modeling approach but you are right it is over-parameterized. You should definitively send a PR to edit or add a new code block to explain this.

**Comment by AlexAndorra** [comment]
Great, will do ASAP 👍 
(FYI, I just checked and Richard does use the "zero-trick" in the original modeling approach)

**Comment by AlexAndorra** [comment]
Closed by #60
